test/perl/TestClient.pl

```diff
@@ -250,16 +250,16 @@ sub usage {
 #
 # SET TEST
 #
-my $setout = [];
+my $setout = {};
 for (my $i = -2; $i < 3; ++$i) {
-    push(@$setout, $i);
+    $setout->{$i} = 1;
```

Comment:
It's just been pointed out to me this this will not work if `$i` is an object, because it will get serialised to its string representation when used as the hash key like this.

Reply:
Do you have a follow-up to this?